### PR TITLE
Add AWS monitoring docs to the observability guide

### LINKED ISSUE #2546: [REQUEST]: Edit and publish Amazon Kinesis documentation
### Description

We need to:

* Edit and possibly restructure the content created by the development team.
* Decide where the content fits into the observability guide.
* Publish the new content at elastic.co.

### Resources

Related issue (internal folks only): https://github.com/elastic/obs-infraobs-team/issues/864 contains a link to source content for this document.

### Collaboration

Other (please describe below in the Point of contact section).

### Point of contact.

**Main contact:** @aspacca 

**Stakeholders:** @girodav

### PR DESCRIPTION
**Pull request by dedemorton**
# Description

Closes #2546 #2547 #2556 #2557

Adds AWS monitoring docs to the observability guide. For the first pass, we will add an AWS monitoring container to the navigation. The final goal is to have a cloud monitoring section where other cloud platforms are represented.

Preview link: https://observability-docs_2564.docs-preview.app.elstc.co/guide/en/observability/master/monitor-amazon-web-services.html

## To Do
- [x] Add raw content created by the development team. Format using asciidoc. No edits just formatting fixes.
  - [x] kinesis
  - [x] s3
  - [x] ec2
  - [x] sqs
- [x] Add links that were stripped out by the copy/paste (as needed--some might be extra)
- [x] Make sure all screen captures have been copied over from the original docs (as needed--remove extraneous slides)
- [x] Check/fix usage of 3rd party brand names.
- [x] Add an introduction to container topic about AWS monitoring.
- [x] Identify gaps or places where the content does not meet our objectives.
- [x] Edit the content to improve readability and conform to Elastic style.
  - [x] For EC2 fix the format of the detailed descriptions that appear after each table.  Might be better to focus on an English description of what user is trying to understand and then list the fields afterwards. I feel we are burying the lede here.
- [x] Make sure the content works well alongside other content about AWS integrations
- [x] Open review.
- [ ] Solicit feedback from internal users (solution architects, support, etc) on usefulness of the content.
- [ ] After merging this PR, consider updating the links in the integrations documentation that currently point to the onboarding guide. (This content is not ideal because it really only works for users who have never added an integration: `For step-by-step instructions on how to set up an integration, see the [Getting started](https://www.elastic.co/guide/en/welcome-to-elastic/current/getting-started-observability.html) guide`)

## Open issues and questions:

- [x] What do folks think of the quick guides? (This is an alternative to sending users to a guide that provides generic info.)
**_Decision_:** We'll keep them and make sure they are consistent with info/process we show in Kibana.
- [x] Best practice is to use inline links to 3rd party docs sparingly to avoid rabbit holing users and to limit dependencies on URLs that may change over time. I've kept the links that I feel are important. Have I removed any inline links that you feel are critical for users?
**_Decision_:** Add links that are generic (future proofed).
- [x] I am very concerned about duplicating the exported field reference information (under "metrics to watch"). This content may get out of sync with the integration packages over time. I can already see that some of the field names don't match (for example, aws.kinesis.metrics.GetRecords_Success.sum). Besides the extra work involved, we'll need to know when theses field names change. What is the plan for keeping this documentation up-to-date as we add new fields or field names change? Historically we have not done a good job of tracking these changes. We need a process.
**_Decision_:** We will remove the summary tables where we duplicate the reference docs, but keep the parts where we provide useful guidance to help users interpret specific fields
- [x] The EC2 topic does a pretty good job of explaining why users should watch specific metrics, but the other topics don't provide any real guidance. Can I get more input about why users should watch the metrics listed in those topics? Otherwise I'm not sure how much value this content adds.
**_Decision_:** We will add this info later after publishing.
- [x] We list the metrics to watch, but don't really tell users how to view them in the UI. Should we say more? For fields that surface in a pre-built visualization, shouldn't we be using the label that users see rather than the field name?
**_Decisio_n:** We should do this as a future enhancement to this content (shouldn't delay publishing this content).
- [x] Do we want to mention Elastic Serverless Forwarder anywhere? I'm thinking specifically of the integrations that don't collect logs.
**_Decision_:** No, not necessary.
- [x] For S3, I've removed the detailed steps and screen captures about enabling request metrics because it's better to point to the AWS docs to "future proof" the content (even the AWS docs don't include screen captures). We have some docs that do...and they are probably already stale. Do you have any concerns about leaving out the screen caps?
**_Decision:_** There was some concern about removing these steps, but we're going to assume that users know their cloud platforms better than we do and will feel comfortable using their platform docs. All links should be as generic as possible, and we should do validation on the links.
- [x] For SQS, we say, "Amazon SQS does not automatically delete the messages in the queue and therefore the consumer is responsible for the deletion of the message after consumption." Why does the user need to know this? Do they need to do anything to delete the messages from the queue, or does the integration handle that? I wonder if the info about the queue really belongs in the intro. 
**_Decision:_** Remove this statement.
- [x] For SQS, why do users need to know about the visibility timeout? Do they to change the default?
**_Decision:_** More investigation required. (Added a comment to the review)

## Future enhancements
- [ ] Provide more detail about why users should watch specific metrics (similar to what we do in the EC2 topic).
- [ ] Where we list the metrics to watch, we should tell users how to view them in the UI.

### CODE REVIEW

**Comment by dedemorton** [comment]
@girodav @bturquet @aspacca How do you want to proceed with feedback on this content? You could read through my list of questions and review the content for your areas with those questions in mind, or we could meet to discuss my questions and concerns.

I'd like to get this PR unstalled, but don't want to move forward until I get some validation that I'm on the right track.

**Comment by girodav** [comment]
Hi @dedemorton, apologies for the late reply. Both @aspacca and myself have been quite busy lately but this issue is in our team's radar.

> You could read through my list of questions and review the content for your areas with those questions in mind, or we could meet to discuss my questions and concerns.

I think having async reviews would be better. We span 3 timezones so it wouldn't be feasible to have a meeting I think.

**Comment by dedemorton** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by dedemorton** [comment]
@aspacca @girodav This PR has been open for awhile. I'm blocked on this until I get a response to my questions/concerns (listed in https://github.com/elastic/observability-docs/pull/2564#issue-1560653768) plus a full review.

I will fix the conflicts to trigger another build, but I have a bunch of other high priority work for upcoming "checkpoints" and would like to get this PR sorted before I'm too far down that rabbit hole.

What do you want to do with this PR? Put it in the freezer (keep it stalled indefinitely)? Get it reviewed and merged?

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-kinesis.asciidoc
Comment (dedemorton) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-kinesis.asciidoc:8]: Add generic link to info about the service. Make sure the link is "future proof" (unlikely to change)

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-ec2.asciidoc
Comment (dedemorton) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-ec2.asciidoc:8]: Add generic link to info about the service. Make sure the link is "future proof" (unlikely to change): https://docs.aws.amazon.com/ec2/index.html

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-s3.asciidoc
Comment (dedemorton) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-s3.asciidoc:8]: Add generic link to info about the service. Make sure the link is "future proof" (unlikely to change)

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc
Comment (dedemorton) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc:12]: Add generic link to info about the service. Make sure the link is "future proof" (unlikely to change)

**Review by dedemorton — COMMENTED** [review]
Reminders to add a generic link to the docs about each monitored service

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-ec2.asciidoc
Comment (dedemorton) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-ec2.asciidoc:85]: Remove this table and point to the fields reference in the integration's overview page instead.

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc
Comment (dedemorton) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc:29]:
```suggestion
```

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc
Comment (dedemorton) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc:23]: @aspacca Can you take a moment to look at this paragraph and let me know if it belongs here. I think it's not clear why the user needs to know this or what they need to do.

**Comment by dedemorton** [comment]
There's still an open question for @aspacca but otherwise this content is ready for approval review.

**Comment by dedemorton** [comment]
@elastic/obs-docs I'm not keen on adding yet another top-level entry to the obs guide nav, but when I put this under Infrastructure monitoring, things get weird. You lose the mini TOC on the pages, and the heading fonts get ridiculously small.

WDYT? A or B? Alternate ideas?

## A: Same level as Infrastructure monitoring
image

## B: Nested below
image

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc
Comment (aspacca) [reply docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc:20]: @kaiyan-sheng can you look at it? thanks

**Comment by colleenmcginnis** [comment]
>You lose the mini TOC on the pages, and the heading fonts get ridiculously small.

Are these the only reasons not to nest this section under "Infrastructure monitoring"? These are the results of bugs in the build system that will be fixed in the long term. I'd advise against making decisions about information architecture based on temporary/unintentional limitations of the tooling.

**Comment by dedemorton** [comment]
@colleenmcginnis Good points. 
>Are these the only reasons not to nest this section under "Infrastructure monitoring"?

No, the tooling limitations are just part of the problem. From an IA perspective, I want to figure out how much nesting we should allow (encourage?) in the TOC, and how many top-level entries are acceptable. We already have a lot of top-level entries in our navigation structure, but deeply nested nav structures can also be a problem for tooling and usability (they require extra clicks).

Here's the full logical hierarchy of where the AWS topics might appear, but I think this is too many levels deep: Infrastructure monitoring > Cloud monitoring > AWS monitoring > Monitor XYZ

It's also not feasible right now given that we only have AWS monitoring content. Until we build out content for other cloud providers and for other types of infra monitoring (like Container monitoring), I think it would be kind of weird to have a deeply nested structure.

When I look on the Elastic website, I see that we don't really distinguish between the types of infrastructure monitoring that we support. We describe everything as "Infrastructure monitoring".

image

Another source of tension for me is that all of the content we currently have about infrastructure monitoring focuses on UI capabilities rather than end user goals. That's why I suggested "AWS monitoring use cases" before, but now that I'm looking at this again, maybe we just need something simple like this:

image

WDYT? Is it weird to shift from UI capabilities to user goals? In an ideal world, all of the content would be ready to publish and user-centered, but that's not the world we're living in. :-)

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc
Comment (kaiyan-sheng) [reply docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc:20]: @dedemorton This is not related to the metric collection so we need to remove it.

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc
Comment (kaiyan-sheng) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc:21]: Same here, not related to metrics.

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc
Comment (kaiyan-sheng) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc]: Same here, not related to metrics.

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc
Comment (kaiyan-sheng) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc:28]: Same here, not related to metrics.

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc
Comment (dedemorton) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc:24]:
```suggestion
```

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc
Comment (dedemorton) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc:32]:
```suggestion
```

File: docs/en/shared/integrations-quick-guide.asciidoc
Comment (dedemorton) [inline docs/en/shared/integrations-quick-guide.asciidoc:32]: Added because clicking this button to open the flyout does not work in the latest snapshot for 8.8

**Comment by dedemorton** [comment]
@aspacca @girodav @bturquet @kaiyan-sheng Can I get final review/approval from someone so we can merge this soon? Thanks!

**Comment by dedemorton** [comment]
@elasticmachine run elasticsearch-ci/docs

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-ec2.asciidoc
Comment (kaiyan-sheng) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-ec2.asciidoc:41]: Do we need to mention logs here?

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-s3.asciidoc
Comment (kaiyan-sheng) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-s3.asciidoc:38]: Do we need to mention logs here?

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-ec2.asciidoc
Comment (dedemorton) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-ec2.asciidoc:41]:
```suggestion
To collect EC2 metrics, you typically need to install the Elastic
```

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-s3.asciidoc
Comment (dedemorton) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-s3.asciidoc:38]:
```suggestion
To collect S3 metrics, you typically need to install the Elastic
```

Comment (dedemorton) [reply docs/en/observability/cloud-monitoring/aws/monitor-amazon-s3.asciidoc:38]: ditto...removed!

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-ec2.asciidoc
Comment (kaiyan-sheng) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-ec2.asciidoc:55]: Sorry I found more places mentioning logs.

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-s3.asciidoc
Comment (kaiyan-sheng) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-s3.asciidoc:53]: same here

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc
Comment (kaiyan-sheng) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc:50]: SQS?

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-ec2.asciidoc
Comment (dedemorton) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-ec2.asciidoc:55]:
```suggestion
{agent} is currently the preferred way to add EC2 metrics. For other
```

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-s3.asciidoc
Comment (dedemorton) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-s3.asciidoc:53]:
```suggestion
{agent} is currently the preferred way to add S3 metrics. For other
```

Comment (dedemorton) [reply docs/en/observability/cloud-monitoring/aws/monitor-amazon-s3.asciidoc:53]: I think it's probably OK to mention that Elastic Agent can collect logs and metrics, but I'll remove the mention of logs in case it's confusing to users.

File: docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc
Comment (dedemorton) [inline docs/en/observability/cloud-monitoring/aws/monitor-amazon-sqs.asciidoc:50]:
```suggestion
For example, to see an overview of your SQS metrics in {kib}, go to
```

**Review by kaiyan-sheng — APPROVED** [review]

**Pull request merged**
